cscs-checks/prgenv/affinity_check.py

```diff
@@ -21,166 +35,520 @@ def __init__(self, variant):
+            self.cpu_set -= cpu_sibilings
+
+
+@rfm.simple_test
+class OneThreadPerCoreOpenMPnomultithread(OneThreadPerCoreOpenMP):
+    '''Only one cpu per core booked.'''
 
-@rfm.parameterized_test(['alternate_socket_filling'],
-                        ['consecutive_socket_filling'],
-                        ['single_task_per_socket_omp'],
-                        ['single_task_per_socket_omp_multithread'])
-class SocketDistributionTest(AffinityTestBase):
-    def __init__(self, variant):
-        super().__init__(variant)
-        self.descr = 'Checking distribution of ranks and threads over sockets.'
-        self.valid_systems = ['daint:mc', 'dom:mc']
-        self.cases = {
-            'alternate_socket_filling': {
-                'ref_daint:mc': 'alternate_socket_filling.txt',
-                'ref_dom:mc': 'alternate_socket_filling.txt',
-                'num_tasks': 36,
-                'num_cpus_per_task': 1,
-                'num_tasks_per_socket': 18,
-                'multithreading': False,
-                'cpu-bind': None,
-            },
-            'consecutive_socket_filling': {
-                'ref_daint:mc': 'consecutive_socket_filling.txt',
-                'ref_dom:mc': 'consecutive_socket_filling.txt',
-                'num_tasks': 36,
-                'num_cpus_per_task': 1,
-                'num_tasks_per_socket': None,
-                'multithreading': False,
-                'cpu-bind': 'rank',
-            },
-            'single_task_per_socket_omp': {
-                'ref_daint:mc': 'single_task_per_socket_omp.txt',
-                'ref_dom:mc': 'single_task_per_socket_omp.txt',
-                'num_tasks': 2,
-                'num_cpus_per_task': 18,
-                'num_tasks_per_socket': 1,
-                'multithreading': False,
-                'cpu-bind': None,
-            },
-            'single_task_per_socket_omp_multithread': {
-                'ref_daint:mc': 'single_task_per_socket_omp_multithread.txt',
-                'ref_dom:mc': 'single_task_per_socket_omp_multithread.txt',
-                'num_tasks': 2,
-                'num_cpus_per_task': 36,
-                'num_tasks_per_socket': 1,
-                'multithreading': True,
-                'cpu-bind': None,
-            },
+    def __init__(self):
+        super().__init__()
+        self.descr = 'Pin one OMP thread per core.'
+        self.use_multithreading = False
+
+        # System-dependent settings here
+        self.system = {}
+
+    @property
+    def ncpus_per_task(self):
+        return self.num_omp_threads
+
+    @rfm.run_before('sanity')
+    def assert_aff_set_length(self):
+        '''Only 1 CPU pinned per thread.'''
+        if not all(len(aff_set) == 1 for aff_set in self.aff_cpus):
+            raise SanityError('incorrect affinity set')
+
+
+@rfm.simple_test
+class OneThreadPerSocketOpenMP(AffinityOpenMPBase):
+    '''Pin each OMP thread to a different socket.'''
+    parameter('omp_bind', ['socket'])
```

Comment:
I get `libgomp: Invalid value for environment variable OMP_PLACES`
According to https://www.openmp.org/spec-html/5.0/openmpse53.html, it should be:
```suggestion
    parameter('omp_bind', ['sockets'])
```